test/test.js

```diff
@@ -70,6 +70,35 @@ test('linting of incorrect Markdown file fails', async t => {
   }
 });
 
+test('linting of incorrect Markdown file fails prints issues as json', async t => {
+  try {
+    await execa('../markdownlint.js',
+      ['--config', 'test-config.json', 'incorrect.md', '--json'],
+      {stripFinalNewline: false});
+    t.fail();
+  } catch (error) {
+    t.is(error.stdout, '');
+    const issues = JSON.parse(error.stderr);
+    t.is(issues.length, 8);        
+    const expected = {
+      fileName: "incorrect.md",
+      lineNumber: 1,
+      ruleNames: [
+        "MD002",
+        "first-heading-h1",
+        "first-header-h1",
+      ],
+      ruleDescription: "First heading should be a top-level heading",
+      ruleInformation: "https://github.com/DavidAnson/markdownlint/blob/v0.23.1/doc/Rules.md#md002",
```

Comment:
The version number here will break in the future - what I do in the library is put `v0.0.0` in the test and run a quick RegExp replace in the test before comparing.